dated_entries_group.py

```diff
@@ -50,11 +47,10 @@ def __new__(cls, string: str):
         # Check if an instance for the given date already exists
         if string in cls._instances:
             return cls._instances[string]
-        else:
-            # If not, create a new instance
-            instance = super(Date, cls).__new__(cls)
-            cls._instances[string] = instance
-            return instance
+        # If not, create a new instance
```

Comment:
I understand your intention here. However, even if else is redundant here (since we have return above), it makes the code a little bit more readable in my opinion. What do you think?

Reply:
sure, It's not matter of live/death - yet I really like the "fast return" pattern, there are a lot of examples  https://realpython.com/python-return-statement/#using-return-with-conditionals :)